src/lib/comms/include/sol-http-server.h

```diff
@@ -68,6 +68,16 @@ struct sol_http_server;
 struct sol_http_request;
 
 /**
+ * @struct sol_http_response_sse
+ * @brief Opaque handler for response using server-sent-event.
+ *
+ * This response is create with @c sol_http_server_send_response_sse
+ * and data can be given using @c sol_http_server_reponse_sse_send_data,
+ * to delete it use @c sol_http_server_reponse_ss2_del.
+ */
+struct sol_http_response_sse;
```

Comment:
I do not like to name this so specifically after `sse`. Usually dynamic pages, a big bulk of internet content, is not "one go" responses, but rather they are created interactively. Old PHP, Perl or so called CGI-BIN were like that, you started to reply with "print" to stdout (which was redirected to socket), then you did some DB query, then you printed each line, and so on. Contrast this with the API we support at the moment, where you must create a fixed content and send it in one go.

If you have fixed amount of memory, like 5Mb, you cannot serve a big content using the "fixed" response, since you'd end without memory after 5Mb. However with dynamic responses you progressively dispatch chunks and 5Mb will be enough to deliver a 100Mb contents extracted from DB (see above).

Then we can call this "dynamic" or "progressive". @lpereira any suggestion of a good name?

`sol_http_response_progressive`
`sol_http_response_dynamic`

maybe progressive represents best.